x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -182,9 +182,7 @@ public void searchProfile(SearchProfilesRequest request, ActionListener<SearchPr
                         "user_profile.user.full_name",
                         "user_profile.user.full_name._2gram",
                         "user_profile.user.full_name._3gram",
-                        "user_profile.user.display_name",
-                        "user_profile.user.display_name._2gram",
-                        "user_profile.user.display_name._3gram"
+                        "user_profile.user.email"
```

Comment:
I think we would want the email to also be of type `search_as_you_type` rather than `text`, and update this query builder in consequence.
Happy to defer it to a follow-up if you raise a tracking issue for it.

Reply:
I suspect we might still need to improve it. But since email field has its own `email` analyzer, `search_as_you_type` does not seem to offer any extra advantage over `text`. For example, an email `foo.bar@example.org` with text mapping can be matched with `foo`, `bar`, `foo ba`, `fo.bar`, `ba.foo`, `example.org` etc which seem to have sufficient coverage. Anyway, I added it as a future item in the agenda so we can revisit it later.